### ISSUE TITLE
WAL support

### ISSUE THREAD

**Issue by rkusa** [rkusa/sqlite-vfs#5]
Most WAL tests are already green, only the following are remaining to be fixed:

- [ ] wal5.test
- [ ] wal6.test
- [ ] walro.test
- [ ] walro2.test
- [ ] walthread.test
- [ ] walvfs.test

**Comment by danthegoodman1** [comment]
curious on try this out, where are the tests that are failing coming from? (or rather how can I run the tests to see if modifications make them pass)?

Edit: nvm https://github.com/rkusa/sqlite-vfs/blob/main/Makefile#L3 :)

any intel on how to fix would be a helpful starting point though!

**Comment by rkusa** [comment]
@danthegoodman1 IIRC, all shared memory (calls starting with `shm_`) were related to the WAL implementation, e.g.: https://github.com/rkusa/sqlite-vfs/blob/main/src/lib.rs#L1518

Disclaimer: my use-case of `sqlite-vfs` went away, so I am not using it in any production capacity and cannot vouch for its reliability even outside of the missing WAL support.

**Comment by danthegoodman1** [comment]
Thank you! I don’t think a vfs normally has to manage the shm file though right?

**Comment by danthegoodman1** [comment]
Or maybe that’s not related, but it seems the VCs spec doesn’t require shm_ calls?

**Comment by rkusa** [comment]
@danthegoodman1 I honestly don't remember exactly, so you might be right and WAL was not related to those `shm_` calls. Since I implemented [WAL related parts](https://github.com/rkusa/sqlite-vfs/blob/main/src/lib.rs#L334-L336), it might actually be that I just didn't invest the time to get the test suite green.

**Comment by danthegoodman1** [comment]
Yeah I think it's simpler than that, the VFS is "blind" to the WAL for the most part: it doesn't need any special methods for it IIUC. I think otherwise all of the FFI and registering should be good though, might need some work to get it standalone so it can be imported rather than only being able to use from within rust atm.
